pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -362,10 +362,16 @@ func (c *Controller) AppendWorkloadHandler(f func(*model.WorkloadInstance, model
 
 // GetIstioServiceAccounts implements model.ServiceAccounts operation
 func (c *Controller) GetIstioServiceAccounts(svc *model.Service, ports []int) []string {
+	out := map[string]struct{}{}
 	for _, r := range c.GetRegistries() {
-		if svcAccounts := r.GetIstioServiceAccounts(svc, ports); svcAccounts != nil {
-			return svcAccounts
+		svcAccounts := r.GetIstioServiceAccounts(svc, ports)
+		for _, sa := range svcAccounts {
```

Comment:
How can one service reside in two registries?

Reply:
when user deploy the same service, say httpbin, in both clusters. Istio will distribute the union of the endpoints to the envoy, and envoy so that can do load balancing across different service registries.

but the SPIFFE san is only partial. This can cause mTLS connection failure issue.

Reply:
What do you mean by "the SPIFFE san is only partial"?

Reply:
httpbin service, cluster1: default/sa1, default/sa2; cluster2: default/sa2, default/sa3.

The secure naming should generate [spiffe://td/ns/default/sa1, spiffe://td/ns/default/sa2, spiffe://td/ns/default/sa3]. This is even a bug without different trust domains.

Reply:
See test case for an example.

Reply:
See comment above - trust domain aliases ( for migration, etc ) need to also be returned

Reply:
Also: it is a bit arguable if this is the correct behavior or a security vulnerability. 

We get most configs from the 'config controller' - the remote registries are supposed to only provide endpoints, not security configs, because they may be less trust-worthy.  This is in effect a security config - the endpoint tells the system which service account it has. Can we add a flag at least to disable reading service accounts from remote registries, and start a doc for security review to get to a conclusion on this ?